src/simple.ts

```diff
@@ -0,0 +1,255 @@
+import { defaultAbiCoder } from "@ethersproject/abi";
+import { BytesLike, HexString, isBytesLike, toHex, compare } from "./bytes";
+
+import {
+  MultiProof,
+  makeMerkleTree,
+  isValidMerkleTree,
+  getProof,
+  getMultiProof,
+  processProof,
+  processMultiProof,
+  renderMerkleTree,
+} from "./core";
+
+import { SimpleMerkleTreeOptions, defaultOptions } from "./options";
+import { checkBounds } from "./utils/check-bounds";
+import { throwError } from "./utils/throw-error";
+
+export type SimpleMerkleTreeValue = {
+  value: BytesLike[];
+  treeIndex: number;
+};
+
+export type SimpleMerkleTreeData = {
+  format: string;
+  tree: HexString[];
+  values: SimpleMerkleTreeValue[];
+};
+
+/// @notice Not an actual hash function, just a placeholder
+export function simpleLeafHash(value: BytesLike[]): HexString {
+  return defaultAbiCoder.encode(["bytes32"], value);
+}
+
+export interface MerkleTree<T extends BytesLike[] = BytesLike[]> {
+  root: HexString;
+  dump(): SimpleMerkleTreeData;
+  render(): string;
+  entries(): Iterable<[number, T]>;
+  validate(): void;
+  leafLookup(leaf: T): number;
+  getProof(leaf: number | T): HexString[];
+  getMultiProof(leaves: (number | T)[]): MultiProof;
+  verify(leaf: number | T, proof: HexString[]): boolean;
+  verifyMultiProof(multiproof: MultiProof): boolean;
+}
+
+export class SimpleMerkleTree implements MerkleTree {
+  private readonly hashLookup: { [hash: HexString]: number };
+
+  protected constructor(
+    protected readonly tree: HexString[],
+    protected readonly values: SimpleMerkleTreeValue[],
+    protected readonly options: SimpleMerkleTreeOptions = {}
+  ) {
+    this.options = Object.assign(options, defaultOptions);
```

Comment:
We don't want to override options that are set with the default ones.
```suggestion
    this.options = Object.assign(defaultOptions, options);
```

Reply:
@Amxx's comment is right but there is another (small) issue with `Object.assign`. TypeScript allows the user to pass in `{ sortLeaves: undefined }` and this will override the default. AFAIK the best way to solve this is to write out the full object with each option listed:

```suggestion
    this.options = {
      sortLeaves: options.sortLeaves ?? defaultOptions.sortLeaves,
      /* ... the rest */
    }
```